### PR TITLE
refactor schechter stellar mass sampling

### PR DESCRIPTION
**Pull request by itrharrison**
## Description

Refactors `skypy.galaxy.stellar_mass` to be in `skypy.galaxy._schechter` and hence the sampling of `skypy.galaxy._schechter.schechter_smf` to work in the same manner as `skypy.galaxy._schechter.schechter_lf`.

## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/master/CONTRIBUTING.md)
- [x] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by itrharrison** [comment]
Currently fails due to the dependent_argument of redshift for stellar_mass.schechter_smf_mass.

The function itself has no reason to depend on redshift, but the parameter alpha can be a function of redshift.

Is the way to fix this add the redshift argument as
`def schechter_smf_mass(redshift, m_star, alpha, m_min, m_max, size=None,
                       resolution=1000):`
which will then be unused? Or to remove the `dependent_argument` decorators above the function?

**Review by ntessore — APPROVED** [review]
🎉

File: skypy/galaxy/stellar_mass.py [hunk 1]
```diff
@@ -66,9 +60,19 @@ def schechter_smf(alpha, m_star, x_min, x_max, resolution=100, size=None):
...
+        size = np.shape(redshift)
+
+    # convert m_min, m_max to units of m_star
+    x_min = m_min / m_star
+    x_max = m_max / m_star
 
-    x_sample = schechter(alpha, x_min, x_max, resolution=resolution, size=size)
+    # sample masses
+    m = schechter(alpha, x_min, x_max, resolution, size=size)
+    m *= m_star
```
Comment (rrjbca) [inline skypy/galaxy/stellar_mass.py:76]:
```suggestion
    m = schechter(alpha, x_min, x_max, resolution, size=size, scale=m_star)
```

**Review by rrjbca — APPROVED** [review]
Great work thanks @itrharrison

**Pull request merged**